USING_ELEKTRA.md

```diff
@@ -0,0 +1,252 @@
+#> description
+#> type
+```
+Here we can als see `check/enum/#0-7` which is used by Elektra to assert
+for correct configuration settings. You can query for the values by calling
+`kdb getmeta`
+```sh
+kdb getmeta '/sw/lcdproc/lcdd/#0/current/curses/background' 'check/enum/#0'
+#> red
+```
+This `.../#<number>` notation is the typical array notation which is used by Elektra.
+If you want to query all metadata and see their values you can use this little script:
+```sh
+kdb lsmeta '/sw/lcdproc/lcdd/#0/current/curses/background' \
+| xargs -I% -n1 sh -c 'printf "% = " && kdb getmeta \
+"/sw/lcdproc/lcdd/#0/current/curses/background" "%"'
+#> check/enum = #7
+#> check/enum/#0 = red
+#> check/enum/#1 = black
+#> check/enum/#2 = green
+#> check/enum/#3 = yellow
+#> check/enum/#4 = blue
+#> check/enum/#5 = magenta
+#> check/enum/#6 = cyan
+#> check/enum/#7 = white
+#> default = cyan
+#> description = background color when "backlight" is off
+#> type = string
+```
+In future versions we might include a command which gives you all metadata 
+including their values with a simple command.
+If you try to set the value of the background to an invalid value, Elektra
+will raise an error and prevent a possible misconfiguration:
+```sh
+kdb set '/sw/lcdproc/lcdd/#0/current/curses/background' purple
+#> Using name user/sw/lcdproc/lcdd/#0/current/curses/background
+#> The command kdb set failed while accessing the key database with the info:
+#> Sorry, the error (#121) occurred ;(
+#> Description: validation failed
+#> Reason: Validation of key "user/sw/lcdproc/lcdd/#0/current/curses/background" with string "purple" failed.
+#> Ingroup: plugin
+#> Module: enum
+#> At: <path>/src/plugins/enum/enum.c:218
+#> Mountpoint: user/sw/lcdproc/lcdd/#0/current
+#> Configfile: <path>/.config/LCDd.conf.17145:1549993505.561040.tmp
+```
+
+### Kdb Editor
+
+Another sophisticated possibility to see available options and edit them
+is to use the `kdb editor`. If you call `kdb editor <key>`, a local editor will
+be opened and all keys under the given `<key>` including all metadata will be shown.
+
+```sh
+kdb editor 'spec/sw/lcdproc/lcdd/#0/current/curses/backlight' ni
+##Editor
+#> []
+#>  check/enum/#1 = black
+#>  description = background color when "backlight" is off
+#>  check/enum/#6 = cyan
+#>  check/enum/#5 = magenta
+#>  check/enum/#4 = blue
+#>  check/enum = #7
+#>  default = cyan
+#>  check/enum/#7 = white
+#>  check/enum/#2 = green
+#>  check/enum/#0 = red
+#>  check/enum/#3 = yellow
+#>  type = string
+```
+ 
+As of now the editor is not able to handle multiple namespaces at once so be sure to always use 
+the key of the relevant namespace.
+
+If you want to change a value, the best way is to manually set it with `kdb set`. Afterwards
+the editor can be used to investigate all associated metadata and change the value afterwards.
+
+```sh
+kdb set '/sw/lcdproc/lcdd/#0/current/curses/background' red
+kdb editor 'user/sw/lcdproc/lcdd/#0/current/curses/backlight' ni
```

Comment:
Why not directly edit user/sw/lcdproc/lcdd/#0/current

Reply:
I could also do that but this way it is more consistent in the tutorial. I added a line stating how to see all values

Reply:
The editor is different to use as `kdb set`.